services-core/src/gas_price/priority.rs

```diff
@@ -1,14 +1,36 @@
 use super::GasPriceEstimating;
 use anyhow::{anyhow, Result};
-use std::{future::Future, time::Duration};
+use std::{
+    future::Future,
+    sync::atomic::{AtomicU64, Ordering},
+    time::Duration,
+};
+
+// Errors of an individual estimator are logged as warnings until it has failed this many times in
+// a row at which point they are logged as errors.
+// This is useful to reduce alerts for estimators that sometimes fail individual requests while
+// still getting them when the estimator really goes down.
+const LOG_ERROR_AFTER_N_ERRORS: u64 = 10;
 
 // Uses the first successful estimator.
 pub struct PriorityGasPrice {
-    estimators: Vec<Box<dyn GasPriceEstimating>>,
+    estimators: Vec<Estimator>,
+}
+
+struct Estimator {
+    estimator: Box<dyn GasPriceEstimating>,
+    errors_in_a_row: AtomicU64,
 }
 
 impl PriorityGasPrice {
     pub fn new(estimators: Vec<Box<dyn GasPriceEstimating>>) -> Self {
+        let estimators = estimators
+            .into_iter()
+            .map(|estimator| Estimator {
+                estimator,
+                errors_in_a_row: AtomicU64::new(0),
+            })
```

Comment:
ubernanonit: implement .into() trait for style?

Reply:
I prefer it without if it's local to this file. Less to type, less traits to understand.